src/itkOMEZarrNGFFImageIO.cxx

```diff
@@ -246,6 +246,10 @@ itkToTensorstoreComponentType(const IOComponentEnum itkComponentType)
   }
 }
 
+// Variable template, ensuring that tensorstoreToITKComponentType is evaluated at compile-time.
+template <typename T>
+static constexpr IOComponentEnum toITKComponentType = tensorstoreToITKComponentType(tensorstore::dtype_v<T>);
```

Comment:
We could bypass the troublesome compile-time evaluation of tensorstoreToITKComponentType/tensorstore::DataType::id() by defining the variable template as follows:

```cpp
template <typename T>
static constexpr IOComponentEnum toITKComponentType =
  std::is_same_v<T, int8_t> ? IOComponentEnum::CHAR :
  std::is_same_v<T, uint8_t> ? IOComponentEnum::UCHAR :
  std::is_same_v<T, int16_t> ? IOComponentEnum::SHORT :
  std::is_same_v<T, uint16_t> ? IOComponentEnum::USHORT :
  std::is_same_v<T, int32_t> ? IOComponentEnum::INT :
  std::is_same_v<T, uint32_t> ? IOComponentEnum::UINT :
  std::is_same_v<T, int64_t> ? IOComponentEnum::LONGLONG :
  std::is_same_v<T, uint64_t> ? IOComponentEnum::ULONGLONG :
  std::is_same_v<T, float> ? IOComponentEnum::FLOAT :
  std::is_same_v<T, double> ? IOComponentEnum::DOUBLE :
  IOComponentEnum::UNKNOWNCOMPONENTTYPE;
```

Do we like that?  🤔

Reply:
It is probably better if they fix it. Let's wait a few days?

Reply:
OK, let's wait a few days 👍 By the way, clang-format would make the "is_same_v based" definition from https://github.com/InsightSoftwareConsortium/ITKIOOMEZarrNGFF/pull/68#discussion_r1723604278 look like this:

```cpp
template <typename T>
static constexpr IOComponentEnum toITKComponentType =
  std::is_same_v<T, int8_t>
    ? IOComponentEnum::CHAR
    : std::is_same_v<T, uint8_t>
        ? IOComponentEnum::UCHAR
        : std::is_same_v<T, int16_t>
            ? IOComponentEnum::SHORT
            : std::is_same_v<T, uint16_t>
                ? IOComponentEnum::USHORT
                : std::is_same_v<T, int32_t>
                    ? IOComponentEnum::INT
                    : std::is_same_v<T, uint32_t>
                        ? IOComponentEnum::UINT
                        : std::is_same_v<T, int64_t>
                            ? IOComponentEnum::LONGLONG
                            : std::is_same_v<T, uint64_t>
                                ? IOComponentEnum::ULONGLONG
                                : std::is_same_v<T, float>
                                    ? IOComponentEnum::FLOAT
                                    : std::is_same_v<T, double> ? IOComponentEnum::DOUBLE
                                                                : IOComponentEnum::UNKNOWNCOMPONENTTYPE;

```

Wonderful, isn't is?  😸